### PR TITLE
[DataFrame] Implemented prod, product, added test suite

### PR DESCRIPTION
**Pull request by hsubbaraj**
Implemented prod, product
pandas test suite for min, max, cumulative functions, sum, prod, product
added an additional missing parameter to sum

### CODE REVIEW

File: python/ray/dataframe/test/test_analytics.py [hunk 1]
```diff
@@ -0,0 +1,353 @@
+from __future__ import absolute_import
```
Comment (devin-petersohn) [inline python/ray/dataframe/test/test_analytics.py]: We should remove this file.

**Review by p-yang — CHANGES_REQUESTED** [review]

File: python/ray/dataframe/test/test_dataframe.py [hunk 2]
```diff
@@ -2205,18 +2209,14 @@ def test_pow():
...
 
-def test_prod():
-    ray_df = create_test_dataframe()
-
-    with pytest.raises(NotImplementedError):
-        ray_df.prod(None)
-
+@pytest.fixture
+def test_prod(ray_df, pandas_df):
+    assert(ray_df_equals_pandas(ray_df.prod().sort_index(), pandas_df.prod().sort_index()))
```
Comment (devin-petersohn) [inline python/ray/dataframe/test/test_dataframe.py]: Please remove `sort_index()` from each side of this. The index should come out the same.

File: python/ray/dataframe/test/test_dataframe.py [hunk 3]
```diff
@@ -2205,18 +2209,14 @@ def test_pow():
...
 
-def test_product():
-    ray_df = create_test_dataframe()
 
-    with pytest.raises(NotImplementedError):
-        ray_df.product()
+@pytest.fixture
+def test_product(ray_df, pandas_df):
+    assert(ray_df_equals_pandas(ray_df.product().sort_index(), pandas_df.product().sort_index()))
```
Comment (devin-petersohn) [inline python/ray/dataframe/test/test_dataframe.py]: Same as above regarding `sort_index()`.

**Review by devin-petersohn — COMMENTED** [review]
Looks great! Just a couple of minor changes.

**Review by devin-petersohn — COMMENTED** [review]
After you rebase and replace the newlines, looks good!

**Review by devin-petersohn — APPROVED** [review]

**Comment by hsubbaraj** [comment]
Prod returns a Series, but I was using a dataframe check instead. This commit should resolve the issue

**Comment by devin-petersohn** [comment]
Merged, thanks @hsubbaraj!

**Pull request merged**